pkg/piperutils/fileUtils_test.go

```diff
@@ -81,20 +82,16 @@ func TestCopy(t *testing.T) {
 	})
 }
 
-func runInTempDir(t *testing.T, nameOfRun, tempDirPattern string, run func(t *testing.T)) {
-	dir, err := ioutil.TempDir("", tempDirPattern)
-	if err != nil {
-		t.Fatal("Failed to create temporary directory")
-	}
-	oldCWD, _ := os.Getwd()
-	_ = os.Chdir(dir)
-	// clean up tmp dir
-	defer func() {
-		_ = os.Chdir(oldCWD)
-		_ = os.RemoveAll(dir)
-	}()
+func runInTempDir(t *testing.T, nameOfRun string, run func(t *testing.T)) {
+	t.Run(nameOfRun, func(t *testing.T) {
+		dir := t.TempDir()
+		oldCWD, _ := os.Getwd()
+		_ = os.Chdir(dir)
 
-	t.Run(nameOfRun, run)
+		run(t)
+
+		_ = os.Chdir(oldCWD)
```

Comment:
Why don't you keep this in the defer?

Reply:
You are right. I should have put it in `t.Cleanup` like so

```go
func runInTempDir(t *testing.T, nameOfRun string, run func(t *testing.T)) {
	t.Run(nameOfRun, func(t *testing.T) {
		dir := t.TempDir()
		oldCWD, _ := os.Getwd()
		_ = os.Chdir(dir)
		t.Cleanup(func() {
			_ = os.Chdir(oldCWD)
		})

		run(t)
	})
}
```

This guarantees that the working directory is changed back `oldCWD` even if the test panics.